[SPARK-28792][SQL][DOC] Document CREATE DATABASE statement in SQL Reference.

sharangk opened the pull request:
### What changes were proposed in this pull request?
Document CREATE DATABASE statement in SQL Reference Guide.

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes.

### Before:
There was no documentation for this.
### After:
![image](https://user-images.githubusercontent.com/29914590/65037831-290e2900-d96c-11e9-8563-92e5379c3ad1.png)
![image](https://user-images.githubusercontent.com/29914590/64858915-55f9cd80-d646-11e9-91a9-16c52b1daa56.png)



### How was this patch tested?
Manual Review and Tested using jykyll build --serve

sharangk committed:
[SPARK-28792][SQL][DOC] Document CREATE DATABASE statement in SQL Reference

# Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

## How was this patch tested?
This is just a documentation change, verified manually.

sharangk commented:
@dilipbiswal , @gatorsmile , Please review. Thanks

huaxingao commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,39 @@ license: |
+CREATE (DATABASE|SCHEMA) [IF NOT EXISTS] database_name
+  [COMMENT database_comment]
+  [LOCATION database_directory]
+  [WITH DBPROPERTIES (property_name=property_value, ...)]
```
@dilipbiswal @gatorsmile 
I am wondering if we should define a standard for the syntax format. Should we have a white space between the braces, =, |, ... and the text? I currently put a white space because I saw a couple of sql references have these standard. I am OK with either having white space or no white space, but I am thinking we may need to define a standard and write out the standard explicitly in the umbrella jira so everybody can follow.  Also, shall we have 4 white space indentation for the 2nd line? I am ok with either 2 or 4, but prefer everybody to follow the same way.

dilipbiswal replied:
@huaxingao About while spacing , i am +1 on this. About the indentation on the 2nd line, i don't have a strong preference. @gatorsmile do you have a preference here ?

sharangk replied:
Updated white spaces wherever required. Thanks

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,39 @@ license: |
+{% highlight sql %}
+-- Create database `customer_db`. This throws exception if database with name customer_db already exists.
+CREATE DATABASE customer_db;
+-- Create database `customer_db` only if database with same name doesn't exist.
```
Between two tests, lets have a empty line ?

sharangk replied:
Done. Thanks

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,39 @@ license: |
+-- Create database `customer_db` only if database with same name doesn't exist.
+CREATE DATABASE IF NOT EXISTS customer_db;
+-- Create database `customer_db` only if database with same name doesn't exist with `Comments`, `Specific Location` and `Database properties`.
+CREATE DATABASE IF NOT EXISTS customer_db COMMENT 'This is customer database' LOCATION '/user' WITH DBPROPERTIES (ID=001, Name='John');
```
Indent this ?

sharangk replied:
done

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,39 @@ license: |
+### Parameters
+- ##### ***IF NOT EXISTS***:
+Creates a database with the given name if it doesn't exists. If a database with the same name already exists, nothing will happen.
+- ##### ***LOCATION***:
```
Since this is an optional clause, should we specify what happens when this is not specified ? The database is created in the default warehouse directory ? Could you please check ?

sharangk replied:
Updated. Thanks

dilipbiswal commented:
@sharangk  Thanks for working on it. Can we please re-word some descriptions ? It sounds similar to some other docs.

sharangk commented:
Thanks for the timely review. I will work the comments and resubmit the PR.

sharangk committed:
[SPARK-28792][SQL][DOC] Document CREATE DATABASE statement in SQL Reference

# Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

## How was this patch tested?
This is just a documentation change, verified manually.

sharangk commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,39 @@ license: |
+Description for the Database.
+- ##### ***WITH DBPROPERTIES***:
+Properties for the database in key-value pair.
+
```
Formatted based on #25523 . Thanks

sharangk commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,39 @@ license: |
+{% endhighlight %}
+
+
+### Related statements.
```
Updated. Thanks

sharangk commented:
Handled the comments. Thanks @dilipbiswal @huaxingao for timely review.
@gatorsmile , Could you please have a look at the changes.

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,58 @@ license: |
+    <dd>Path of the file system in which database to be created. If the specified path does not exist in the underlying file system, this command creates a directory with the path. If location is not specified, database will be created in default warehouse directory.</dd>
+
+    <dt><code><em>database_comment</em></code></dt>
+    <dd>Description for the database.</dd>
```
Tiny nit : Specifies the description for the database.

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,58 @@ license: |
+    <dd>Description for the database.</dd>
+
+    <dt><code><em>WITH DBPROPERTIES (property_name=property_value [ , ...])</em></code></dt>
+    <dd>Properties for the database in key-value pair.</dd>
```
Tiny nit: add Specifies

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,58 @@ license: |
+
+### Examples
+{% highlight sql %}
+-- Create database `customer_db`. This throws exception if database with name customer_db already exists.
```
Can we put the 2nd line under a comment  like 
```
-- Create database `customer_db`. This throws exception if database with name customer_db
-- already exists.
```

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,58 @@ license: |
+-- Create database `customer_db` only if database with same name doesn't exist.
+CREATE DATABASE IF NOT EXISTS customer_db;
+
+-- Create database `customer_db` only if database with same name doesn't exist with `Comments`,`Specific Location` and `Database properties`.
```
Can we put the 2nd line under a comment  like 
```
-- Create database `customer_db` only if database with same name doesn't exist with
-- `Comments`,`Specific Location` and `Database properties`.
```

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,58 @@ license: |
+CREATE DATABASE IF NOT EXISTS customer_db;
+
+-- Create database `customer_db` only if database with same name doesn't exist with `Comments`,`Specific Location` and `Database properties`.
+CREATE DATABASE IF NOT EXISTS customer_db COMMENT 'This is customer database' LOCATION '/user' WITH DBPROPERTIES (ID=001, Name='John');
```
lets indent the 2nd line so that it looks continuation of first line like :
```
CREATE DATABASE IF NOT EXISTS customer_db COMMENT 'This is customer database' LOCATION '/user' 
  WITH DBPROPERTIES (ID=001, Name='John');
```

dilipbiswal commented:
@sharangk Mostly looks good. I have put some minor comments.

sharangk committed:
updated review comments

sharangk commented:
@dilipbiswal , Thanks for the review. Resubmitted the PR with updates. Please have a look. Thanks !!!

AmplabJenkins commented:
Can one of the admins verify this patch?

dilipbiswal commented on docs/sql-ref-syntax-ddl-create-database.md:
```diff
@@ -19,4 +19,61 @@ license: |
+    <dd>Creates a database with the given name if it doesn't exists. If a database with the same name already exists, nothing will happen.</dd>
+
+    <dt><code><em>database_directory</em></code></dt>
+    <dd>Path of the file system in which database to be created. If the specified path does not exist in the underlying file system, this command creates a directory with the path. If location is not specified, database will be created in default warehouse directory.</dd>
```
@sharangk in which the specified database is to be created ?

sharangk committed:
updated sentence

sharangk commented:
@dilipbiswal , Thanks. Handled the review comments.

dilipbiswal commented:
LGTM
cc @gatorsmile

gatorsmile commented:
ok to test

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/15960/
Test PASSed.

gatorsmile committed:
Update sql-ref-syntax-ddl-create-database.md

SparkQA commented:
**[Test build #110828 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/110828/testReport)** for PR 25595 at commit [`0225553`](https://github.com/apache/spark/commit/0225553e7047cec270e8dba8fda983c5cb6920a5).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/15961/
Test PASSed.

SparkQA commented:
**[Test build #110828 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/110828/testReport)** for PR 25595 at commit [`0225553`](https://github.com/apache/spark/commit/0225553e7047cec270e8dba8fda983c5cb6920a5).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/110828/
Test PASSed.

gatorsmile commented:
Thanks! Merged to master.
